core/mo-services-impl/ccsds-com-impl/src/main/java/esa/mo/com/impl/provider/ArchiveSyncProviderServiceImpl.java

```diff
@@ -280,9 +280,7 @@ public int numberOfChunks() throws IOException {
 
         public void addObjects(final ArrayList<COMObjectEntity> list) {
             // "addAll()" might not be thread-safe. "add" is for sure!
-            for (COMObjectEntity entity : list) {
-                tempQueue.add(entity);
-            }
+            tempQueue.addAll(list);
```

Comment:
I think the code comment is mistaking thread-safe for atomic but I'm not familiar with type of Queue (LinkedBlockingQueue), might be worth double checking.

Reply:
As I now understand addAll iterate over a given Collection and call add that is's thread-safe.

Reply:
I just checked and the addObjects only seems to be called in one place, so it should be fine.
I remember coding that defensively because it involves threads passing stuff around to each other.
Didn't have too much time back then, so I took a defensive approach to avoid any bugs from coming out of it.

Reply:
HI @CesarCoelho 
Do you want me to rollback this changes?